### PR TITLE
Delegated accounts custom errors

### PR DESCRIPTION
**Pull request by scotthconner**

### CODE REVIEW

File: src/contracts/core/PermissionController.sol [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+     * onlyAccountAdmin
+     *
+     * This modifier is used to ensure that functions can only be accessed
+     * by message senders that are admins for the given account.
+     *
+     */
+    modifier onlyAccountAdmin(address account) {
+        require(accounts[account].admins.contains(msg.sender), CallerNotAdmin());
+        _;
+    }
```
[the remark this answers is not in the document]
Comment (scotthconner) [reply src/contracts/core/PermissionController.sol:32]: false alarm it seems

File: src/contracts/core/PermissionController.sol [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        } else {
+            info.delegatedContractSelectors[delegate][target].remove(selector);
+        }
+
+        emit AccountDelegateStateChange(account, msg.sender, target, selector, delegate, hasPermission); 
+    }
+    
+    //////////////////////////////////////////////////
+    // Public Introspection Interface
+    //////////////////////////////////////////////////
```
Comment (0xrajath) [inline src/contracts/core/PermissionController.sol:93]: Is the idea that the core protocol functions would check one of these view functions to see if it's an admin or if it's a delegate and has the right permissions?

File: src/contracts/core/PermissionController.sol [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        if (hasPermission) {
+            info.delegatedContractSelectors[delegate][target].add(selector);
+        } else {
+            info.delegatedContractSelectors[delegate][target].remove(selector);
+        }
+
+        emit AccountDelegateStateChange(account, msg.sender, target, selector, delegate, hasPermission); 
+    }
+    
+    //////////////////////////////////////////////////
```
Comment (0xrajath) [inline src/contracts/core/PermissionController.sol:91]: This would need a `hasPermission` function too.

**Comment by ypatil12** [comment]
Closing in favor of #870

**Pull request closed without merging**